src/components/CribSheet.jsx

```diff
@@ -0,0 +1,198 @@
+import ActiveCard from './ActiveCard';
+
+class CribSheet extends React.Component {
+  constructor() {
+    super();
+
+    this.activateCrop = this.activateCrop.bind(this);
+    this.personalMode = this.personalMode.bind(this);
+    this.referenceMode = this.referenceMode.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+    this.close = this.close.bind(this);
+  }
+
+  onClose() {
+    this.props.dispatch(toggleDialog(null));
+  }
+
+  close(e) {
+    this.onClose();
+    return Utility.stopEvent(e);
+  }
+
+  activateCrop(e) {
+    this.props.dispatch(setViewerState(SUBJECTVIEWER_STATE.CROPPING));
+    this.close(e);
+  }
+
+  renderReference() {
+    // To be replaced with script keyboards
+    return (
+      <div className="crib-sheet__reference">
+        <h2>Hello World</h2>
+      </div>
+    );
+  }
+
+  personalMode() {
+    this.props.dispatch(toggleReferenceMode(false));
+  }
+
+  referenceMode() {
+    this.props.dispatch(toggleReferenceMode(true));
+  }
+
+  header() {
+    const reference = this.props.referenceMode;
+
+    return (
+      <div className="crib-sheet__header handle">
+        {this.props.user && (
+          <button className={!reference ? 'active-button' : ''} onClick={this.personalMode}>Your Crib Sheet</button>
+        )}
+        <button className={reference ? 'active-button' : ''} onClick={this.referenceMode}>Script References</button>
+        <button className="close-button" onClick={this.close}>X</button>
+        <hr className="plum-line" />
+      </div>
+    );
+  }
+
+  deleteItem(i = this.state.cardIndex) {
+    if (this.props.preferences.preferences && this.props.preferences.preferences.cribsheet) {
+      const cribCopy = this.props.preferences.preferences.cribsheet.slice();
+      cribCopy.splice(i, 1);
+      this.props.preferences.update({ 'preferences.cribsheet': cribCopy }).save();
+      this.forceUpdate();
+    }
+  }
+
+  activateCard(activeCard, cardIndex) {
+    this.props.dispatch(activateCard(activeCard, cardIndex));
+  }
+
+  renderItem(snippet, i) {
+    return (
+      <div className="crib-sheet__item" key={`SNIPPET_${i}`}>
+        <button
+          className="crib-sheet__delete"
+          onClick={this.deleteItem.bind(this, i)}
+        >
+          <i className="fa fa-times" />
+        </button>
+
+        <button
+          className="crib-sheet__card"
+          onClick={this.activateCard.bind(this, snippet, i)}
+        >
+          {snippet.cropUrl && (
+            <img alt="Crib Sheet Snippet" src={snippet.cropUrl} />
+          )}
+          <span>
+            {snippet.name}
+          </span>
+        </button>
+
+      </div>
+    );
+  }
+
+  renderInstructions() {
+    return (
+      <div className="crib-sheet__personal-instructions handle">
+        <span>Use this crib sheet to save snippets for your personal reference.</span>
+        <span>If you&apos;re signed in, the images will be saved throughout your time on this project.</span>
```

Comment:
Note a code issue, but a copy query: I'm not certain if "throughout your time on this project" communicates how this cribsheets actually work?